standalone-metastore/metastore-rest-catalog/src/main/java/org/apache/iceberg/rest/HMSCachingCatalog.java

```diff
@@ -63,6 +69,96 @@ public List<Namespace> listNamespaces(Namespace nmspc) throws NoSuchNamespaceExc
     return hiveCatalog.listNamespaces(nmspc);
   }
 
+  protected void cacheInvalidateInc(TableIdentifier tid) {
+    // This method is intentionally left empty. It can be overridden in subclasses if needed.
+  }
+
+  protected void cacheLoadInc(TableIdentifier tid) {
+    // This method is intentionally left empty. It can be overridden in subclasses if needed.
+  }
+
+  protected void cacheHitInc(TableIdentifier tid) {
+    // This method is intentionally left empty. It can be overridden in subclasses if needed.
+  }
+
+  protected void cacheMissInc(TableIdentifier tid) {
+    // This method is intentionally left empty. It can be overridden in subclasses if needed.
+  }
+
+  protected void cacheMetaLoadInc(TableIdentifier tid) {
+    // This method is intentionally left empty. It can be overridden in subclasses if needed.
+  }
+
+  /**
+   * Gets the metadata file location of a table.
+   *
+   * @param table the table
+   * @return the location of the metadata file, or null if the table does not have a location
+   */
+  protected static String getMetadataLocation(final Table table) {
+    if (table instanceof HasTableOperations tableOps) {
+      final TableOperations ops = tableOps.operations();
+      final TableMetadata meta;
+      if (ops != null && (meta = ops.current()) != null) {
+        return meta.metadataFileLocation();
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public Table loadTable(final TableIdentifier identifier) {
+    final Cache<TableIdentifier, Table> cache = this.tableCache;
+    final HiveCatalog catalog = this.hiveCatalog;
+    final TableIdentifier canonicalized = identifier.toLowerCase();
+    Table cachedTable = cache.getIfPresent(canonicalized);
+    if (cachedTable != null) {
+      final String location = catalog.getTableMetadataLocation(canonicalized);
```

Comment:
can't you use `catalog.loadTable(canonicalized)` and get the location instead of adding new method?